### ISSUE TITLE
`search-replace` function blockers

### ISSUE THREAD

**Issue by frankfarzan** [GoogleContainerTools/kpt#1821]
These docs do not work:

https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/master/functions/go/search-replace

Multiple issues:

1. This runs successfully without validating at least a `by-` flag is specified:

```
» kpt fn run --image gcr.io/kpt-fn/search-replace:unstable .
```

2. In addition, where there are no matches, there's no indication just a blank line printed:

```
$ kpt fn run --image gcr.io/kpt-fn/search-replace:unstable . -- 'by-path=metadata.name' 'put-value=the-deployment'

```

3. Also, the docs here incorrectly refer to `apply-setters`, probably a copy-paste issue.

4. In general, findings are not printed today.

Please do an audit of the usage doc, make sure everything works as expected, corner cases are tested, etc.

**Comment by phanimarupaka** [comment]
> These docs do not work:
> 
> https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/master/functions/go/search-replace
> 
> Multiple issues:
> 
> 1. This runs successfully without validating at least a `by-` flag is specified:
> 
Ack.
> ```
> » kpt fn run --image gcr.io/kpt-fn/search-replace:unstable .
> ```
> 
> 1. In addition, where there are no matches, there's no indication just a blank line printed:
> 
> ```
> $ kpt fn run --image gcr.io/kpt-fn/search-replace:unstable . -- 'by-path=metadata.name' 'put-value=the-deployment'
> ```
As in no output? This was intentional. Other tools like `yq` also does the same rather than throwing an error. The current behavior is correct IMO.
> 
> 1. Also, the docs here incorrectly refer to `apply-setters`, probably a copy-paste issue.
Ack.
> 2. In general, findings are not printed today.
This will be addressed by https://github.com/GoogleContainerTools/kpt/issues/1557
> 
> Please do an audit of the usage doc, make sure everything works as expected, corner cases are tested, etc.

**Comment by phanimarupaka** [comment]
@frankfarzan 1,2,3 issues are resolved. 4 is a general issue with functions output in general and is being tracked here https://github.com/GoogleContainerTools/kpt/issues/1557, should we keep this till it is closed? And the issue description is not very accurate as replace works fine and search also produces results if --results-dir is specified. Issue is with displaying output. Consider changing the description.

**Comment by frankfarzan** [comment]
There's a reference in the book to this issue. Let's leave this open even if the issue is not in the func impl as a signal to the readers of the book.

**Comment by droot** [comment]
These should be addressed with #1961
